src/ReusableRelayBuilder.php

```diff
@@ -0,0 +1,39 @@
+<?php
+namespace Relay;
+
+use ArrayObject;
+use InvalidArgumentException;
+
+class ReusableRelayBuilder
+{
+    protected $resolver;
+
+    public function __construct($resolver = null)
+    {
+        $this->resolver = $resolver;
+    }
+
+    public function newInstance($queue)
+    {
+        return new ReusableRelay(new RelayFactory(
+            $this->getArray($queue),
+            $this->resolver
+        ));
+    }
+
+    protected function getArray($queue)
+    {
+        if (is_array($queue)) {
+            return $queue;
+        }
+
+        $getArrayCopy = $queue instanceof GetArrayCopyInterface
+            || $queue instanceof ArrayObject;
+
+        if ($getArrayCopy) {
+            return $queue->getArrayCopy();
```

Comment:
Ah so! I like that idea, though it does not apply specifically to the reusability of the relay; that logic pre-exists in the 1.x branch in RelayBuilder. Let's take a look at that separately from this.